### PR TITLE
Use ab_glyph instead of rusttype for font rendering

### PR DESCRIPTION
**Pull request by bnjbvr**
`rusttype` is unmaintained: https://gitlab.redox-os.org/redox-os/rusttype/-/issues/148, and the author has created a new lib called `ab_glyph` that does the same things with a slightly different API. My goal was to remove a dependency on a very old `owned_ttf_parser` in this crate, but turns out that the whole Rust ecosystem relies on `rusttype`, it seems, and this dependency is still in the `Cargo.lock` file because `winit` hence `glutin` hence `glium` rely on it. Anyways, here's the PR if you're interested :-)

### CODE REVIEW

**Review by emilk — APPROVED** [review]
This is great, thank you! I hadn't noticed that `rusttype` is unmaintained. `ab_glyph` looks great too. I see there is now [a way to iterate over all the glyphs in a font](https://docs.rs/ab_glyph/0.2.11/ab_glyph/trait.Font.html#tymethod.codepoint_ids), meaning we no longer need to hard-code the list of characters for the font book (though we still need their names so one can search for glyphs... oh well, that is not a priority right now).

> this dependency is still in the `Cargo.lock` file because `winit` hence `glutin` hence `glium` rely on it

Weird that `winit` depend on `rusttype` 🤦‍♂️ , but most people use `egui` stand-alone and not with `egui_glium`, so this PR is a very nice improvement!

**Comment by emilk** [comment]
Please rebase on `master` to fix the clippy issue!

File: epaint/src/text/font.rs [hunk 1]
```diff
@@ -115,8 +110,9 @@ impl FontImpl {
         }
 
         // Add new character:
-        let glyph = self.rusttype_font.glyph(c);
-        if glyph.id().0 == 0 {
+        use ab_glyph::Font;
```
[the remark this answers is not in the document]
Comment (bnjbvr) [reply epaint/src/text/font.rs]: nice trick, done a few times!

**Comment by emilk** [comment]
Thank you!

**Pull request merged**